### ISSUE TITLE
Better support for remote Docker hosts

### ISSUE THREAD

**Issue by rickardp** [microsoft/vscode-remote-release#3633]
Using development containers against the local Docker host works well, as do remote development over SSH. Working inside development containers on a remote SSH host is a quite painful experience at the moment.

My use case is this:
* My laptop (although decently powerful) where I do all my development, is nowhere near as powerful as the workstation I have in my office. When I have access to a stable Internet connection, I want to do the heavy lifting on this machine.
* I still use local dev containers for some stuff (for example, I am working on a CUDA/machine learning project, as well as small Django/React project currenly. The latter has no problems running on my local Docker host).

These are the problems I have identified:
* Under "Container", I can only see the local containers. I can use a remote SSH host as Docker host, but this requires changing a global setting. I would like this tab to work similar to the "SSH" tab, where I can define multiple hosts simultaneously.
* When setting up the dev container, I need to work with the filesystem on the remote host. It would have been great if I could start out with the remote SSH connection and have the option to switch from there to the container view, just as for the local host ("Rebuild and open in container" command). Now I have to sit with a terminal on the side to manipulate the devcontainer.json, build images, clone the repos, etc.

My proposal:
* Make the "Remote SSH" plugin able to create and use dev containers directly from the remote view, identical to when working with local filesystems.
* Support multiple Docker hosts simultaneously in the container view.

**Comment by chrmarti** [comment]
Makes sense and is tracked as https://github.com/microsoft/vscode-remote-release/issues/2994.

Closing as duplicate.
